src/libs/OptionsListUtils.ts

```diff
@@ -16,47 +24,97 @@ import * as LoginUtils from './LoginUtils';
 import * as ReportActionUtils from './ReportActionsUtils';
 import * as ReportUtils from './ReportUtils';
 import * as TaskUtils from './TaskUtils';
 import * as TransactionUtils from './TransactionUtils';
 import * as UserUtils from './UserUtils';
 
+type Tag = {enabled: boolean; name: string; accountID: number | null};
+
+type Option = {text: string; keyForList: string; searchText: string; tooltipText: string; isDisabled: boolean};
+
+type PayeePersonalDetails = {text: string; alternateText: string; icons: OnyxCommon.Icon[]; descriptiveText: string; login: string; accountID: number};
+
+type CategorySection = {title: string | undefined; shouldShow: boolean; indexOffset: number; data: Option[] | Participant[] | ReportUtils.OptionData[]};
+
+type Category = {
+    name: string;
+    enabled: boolean;
+};
+
+type Hierarchy = Record<string, Category & {[key: string]: Hierarchy & Category}>;
+
+type GetOptionsConfig = {
+    reportActions?: Record<string, ReportAction>;
+    betas?: Beta[];
+    selectedOptions?: Array<Participant | Category>;
+    maxRecentReportsToShow?: number;
+    excludeLogins?: string[];
+    includeMultipleParticipantReports?: boolean;
+    includePersonalDetails?: boolean;
+    includeRecentReports?: boolean;
+    sortByReportTypeInSearch?: boolean;
+    searchInputValue?: string;
+    showChatPreviewLine?: boolean;
+    sortPersonalDetailsByAlphaAsc?: boolean;
+    forcePolicyNamePreview?: boolean;
+    includeOwnedWorkspaceChats?: boolean;
+    includeThreads?: boolean;
+    includeTasks?: boolean;
+    includeMoneyRequests?: boolean;
+    excludeUnknownUsers?: boolean;
+    includeP2P?: boolean;
+    includeCategories?: boolean;
+    categories?: Record<string, PolicyCategory>;
+    recentlyUsedCategories?: string[];
+    includeTags?: boolean;
+    tags?: Record<string, Tag>;
+    recentlyUsedTags?: string[];
+    canInviteUser?: boolean;
+    includeSelectedOptions?: boolean;
+};
+
 /**
  * OptionsListUtils is used to build a list options passed to the OptionsList component. Several different UI views can
  * be configured to display different results based on the options passed to the private getOptions() method. Public
  * methods should be named for the views they build options for and then exported for use in a component.
  */
-
-let currentUserLogin;
-let currentUserAccountID;
+let currentUserLogin: string | undefined;
+let currentUserAccountID: number | undefined;
 Onyx.connect({
     key: ONYXKEYS.SESSION,
-    callback: (val) => {
-        currentUserLogin = val && val.email;
-        currentUserAccountID = val && val.accountID;
+    callback: (value) => {
+        currentUserLogin = value?.email;
+        currentUserAccountID = value?.accountID;
     },
 });
 
-let loginList;
+let loginList: OnyxEntry<Login>;
 Onyx.connect({
     key: ONYXKEYS.LOGIN_LIST,
-    callback: (val) => (loginList = _.isEmpty(val) ? {} : val),
+    callback: (value) => (loginList = Object.keys(value ?? {}).length === 0 ? {} : value),
```

Comment:
If `value` is empty, why not return `value` then instead of an empty object?

Reply:
this is the original implementation I dont want to change it to minimze chances of any regressions